lib/pacemaker/pcmk_sched_allocate.c

```diff
@@ -2393,6 +2392,173 @@ order_first_probes(pe_working_set_t * data_set)
     }
 }
 
+static void
+order_first_probe_then_restart_repromote(pe_action_t * probe,
+                                         pe_action_t * after,
+                                         pe_working_set_t * data_set)
+{
+    GListPtr gIter = NULL;
+    bool interleave = FALSE;
+    pe_resource_t *compatible_rsc = NULL;
+
+    if (probe == NULL || safe_str_neq(probe->task, RSC_STATUS)) {
```

Comment:
Probably not significant, but it would be slightly more efficient to do the string comparison last  (just before pe_set_action_bit())

Reply:
Makes sense.